### PR TITLE
Fix issues with inconsistently renamed D keywords

### PR DESCRIPTION
**Pull request by cbecerescu**

### CODE REVIEW

File: source/dpp/translation/type/package.d [hunk 1]
```diff
@@ -147,7 +147,16 @@ private string translateAggregate(in from!"clang".Type type,
 
             const ns = type.declaration.namespace;
             // no namespace, no problem
-            if(ns.isInvalid) return type.spelling;
+            if(ns.isInvalid) {
+                import std.array : split;
+
+                string[] elems = type.spelling.split(" ");
```
Comment (atilaneves) [inline source/dpp/translation/type/package.d]: s/string[]/auto/

File: source/dpp/runtime/context.d [hunk 2]
```diff
@@ -226,20 +232,34 @@ struct Context {
...
     void declareUnknownStructs() @safe pure {
         foreach(name, _; _fieldStructSpellings) {
-            if(name !in _aggregateDeclarations) {
+            if(isUnknownStruct(name)) {
                 log("Could not find '", name, "' in aggregate declarations, defining it");
-                writeln("struct " ~ name ~ ";");
-                _aggregateDeclarations[name] = true;
+                const spelling = name in _aggregateSpelling ? _aggregateSpelling[name]
+                                                            : name;
+                writeln("struct " ~ spelling ~ ";");
```
[the remark this answers is not in the document]
Comment (cbecerescu) [reply source/dpp/runtime/context.d]: Not sure if the question is just about this part of the code or about the whole idea behind those changes.
This bit of code here (line 259) makes sure that we use the correctly renamed (if renamed) spelling when declaring unknown structures. For example:

```c
void f(struct module *);
```
through dpp becomes
```d
extern(C)
{
   void f(module*) @nogc nothrow;
}

struct module;
```
which will not compile.

The way my changes work: we encounter struct module * in the function declaration, "module" is a keyword so we rename it and map "module" to "module_". When checking for unknown structs, we check if "module" was renamed (it was, so we declare the struct as "struct module_").

[the remark this answers is not in the document]
Comment (atilaneves) [reply source/dpp/runtime/context.d]: The question was about the `writeln` in the code.

[the remark this answers is not in the document]
Comment (cbecerescu) [reply source/dpp/runtime/context.d]: Oh, well, that was there before (previous line 241). The writeln function is not the one from std.stdio. It is actually defined in this exact file (lines 150 and 160) so I guess it actually appends that struct declaration string to the array of lines to be written in the generated D file.

[the remark this answers is not in the document]
Comment (atilaneves) [reply source/dpp/runtime/context.d]: Ah. I missed that, sorry.

**Pull request merged**